src/UKHO.ERPFacade.Common/HttpClients/EesClient.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Net.Http.Headers;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using UKHO.ERPFacade.Common.Authentication;
+using UKHO.ERPFacade.Common.Configuration;
+using UKHO.ERPFacade.Common.Models;
+using UKHO.ERPFacade.Common.Models.CloudEvents;
+
+namespace UKHO.ERPFacade.Common.HttpClients
+{
+    public class EesClient : IEesClient
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<EesClient> _logger;
+        private readonly ITokenProvider _tokenProvider;
+        private readonly IOptions<EESConfiguration> _eesConfiguration;
+
+        public EesClient(HttpClient httpClient, ILogger<EesClient> logger, ITokenProvider tokenProvider, IOptions<EESConfiguration> eesConfiguration)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+            _tokenProvider = tokenProvider;
+            _eesConfiguration = eesConfiguration;
+        }
+
+        public async Task<HttpResponseMessage> Get(string url)
+        {
+            return await _httpClient.GetAsync(url);
+        }
+
+        public async Task<Result> PostAsync(BaseCloudEvent cloudEvent)
```

Comment:
Update this method to return HttpResponseMessage and based on result return Success or Failure from S100UnitOfSaleUpdatedEventPublishingService method

Reply:
Updated as suggested